zerver/tests/test_auth_backends.py

```diff
@@ -1661,8 +1714,77 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
 
         self.email_data = email_data
 
-    def get_account_data_dict(self, email: str, name: str) -> Dict[str, Any]:
-        return dict(email=email, name=name)
+    def get_account_data_dict(self, email: str, name: str, user_avatar_url: str='') -> Dict[str, Any]:
+        return dict(email=email, name=name, avatar_url=user_avatar_url)
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_register_new_invited_user_and_use_github_avatar(self) -> None:
+        """
+        This tests the registration flow in the case where an invitation for the
+        user was generated and it wants to use their Github avatar.
+        """
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        realm = get_realm('zulip')
+        mock_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+
+        iago = self.example_user('iago')
+        do_invite_users(iago, [email], [])
+
+        mocked_github_avatar_image_path = 'zerver/tests/images/img.png'
+
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mock_user_avatar_url)
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup=True)
```

Comment:
It doesn't seem that this actually uses the invitation link; is that intentional?

Reply:
Because in this test I'm not really interested in testing the invitation process. Is just that when an invitation is made, it will create a `prereg_user` and I'm actually just checking if when a user register and already have a `prereg_user` it will fill the `user_avatar_url` correctly for that user. It doesn't really need to use the invitation link for that. But if it wasn't clear, perhaps I should add a comment or change the test name for something more descriptive.